plugins/woocommerce/client/blocks/assets/js/blocks/checkout/inner-blocks/checkout-shipping-address-block/customer-address.tsx

```diff
@@ -21,40 +25,53 @@ const CustomerAddress = () => {
 		setShippingAddress,
 		setBillingAddress,
 		useShippingAsBilling,
-		editingShippingAddress: editing,
-		setEditingShippingAddress: setEditing,
+		editingShippingAddress,
+		setEditingShippingAddress,
 	} = useCheckoutAddress();
 	const { dispatchCheckoutEvent } = useStoreEvents();
 	const [ shouldAnimate, setShouldAnimate ] = useState( false );
 
-	// Forces editing state if store has errors.
-	const { hasValidationErrors, getValidationErrorSelector } = useSelect(
+	const { isInitialized } = useCustomerData();
+
+	const { validationErrors } = useSelect(
 		( select ) => {
-			const store = select( validationStore );
 			return {
-				hasValidationErrors: store.hasValidationErrors(),
-				getValidationErrorSelector: store.getValidationError,
+				validationErrors:
+					select( validationStore ).getValidationErrors(),
 			};
 		},
-		[]
+		[ shippingAddress ]
 	);
 
-	const invalidProps = useMemo( () => {
-		return Object.keys( shippingAddress )
-			.filter( ( key ) => {
-				return (
-					getValidationErrorSelector( 'shipping_' + key ) !==
-					undefined
-				);
-			} )
-			.filter( Boolean );
-	}, [ shippingAddress, getValidationErrorSelector ] );
-
 	useEffect( () => {
-		if ( invalidProps.length > 0 && editing === false ) {
-			setEditing( true );
+		// Check if any shipping field has validation errors
+		const hasValidationErrors = Object.keys( validationErrors ).some(
+			( key ) => {
+				if (
+					key.startsWith( 'shipping_' ) &&
```

Comment:
Do we want to cross-reference this with the list of fields in `ADDRESS_FORM_KEYS` ? if someone adds an error called `shipping_selection` for example, it would trigger this too

Reply:
You could potentially do this by iterating over `ADDRESS_FORM_KEYS` and checking if there's a validation error with `shipping_${ key }` as the name.

Reply:
I checked for the shippingAddress/billingAddress keys, because those have the additional fields as well.